### PR TITLE
Add release steps to Versioning docs

### PR DESCRIPTION
**Pull request by sam-b-rose**
### WHY are these changes introduced?

Adding our release process steps for better transparency around how to create a new Polaris release.

> **Note:** We are working on improving this flow to be less manual and more streamline.

### CODE REVIEW

**Review by alex-page — APPROVED** [review]

File: documentation/Versioning and changelog.md [hunk 1]
```diff
@@ -97,3 +97,35 @@ Because we expose both React components (for which the markup, including class n
 - Changes that do not impact public APIs
 - Non-breaking changes to minimum version of dependencies
 - Breaking changes to private Sass variables, functions, and mixins
+
+## Creating a new release
+
+Follow the following steps to deploy a new version of Polaris:
```
Comment (alex-page) [inline documentation/Versioning and changelog.md]: This has to be done by a @shopify employee. We should probably add that as a disclaimer.

**Comment by sam-b-rose** [comment]
@aveline Thanks for adding the steps for adding Release notes! 💯

**Review by aveline — APPROVED** [review]

File: documentation/Versioning and changelog.md [hunk 2]
```diff
@@ -97,3 +97,48 @@ Because we expose both React components (for which the markup, including class n
...
+
+2. Create a new `polaris-release` branch from `main`
+
+   `git checkout -b polaris-release`
+
+3. Migrate all entries from `UNRELEASED.md` to `CHANGELOG.md` ([see example](https://github.com/Shopify/polaris-react/pull/5254/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed))
+
+4. Commit changelog updates
+
+   `git add . && commit -m "Polaris vX.X.X release"`
```
Comment (aaronccasanova) [inline documentation/Versioning and changelog.md]:
```suggestion
   `git add . && git commit -m "Polaris vX.X.X release"`
```

File: documentation/Versioning and changelog.md [hunk 3]
```diff
@@ -97,3 +97,48 @@ Because we expose both React components (for which the markup, including class n
...
+
+5. Publish the `polaris-release` branch if it isn't already
+
+6. Version and tag the new release
+
+   `yarn version`
+
+7. Push the `polaris-release` branch with tags
+
+   `git push origin --follow-tags`
```
Comment (aaronccasanova) [inline documentation/Versioning and changelog.md]:
```suggestion
   `git push origin polaris-release --follow-tags`
```

File: documentation/Versioning and changelog.md [hunk 4]
```diff
@@ -97,3 +97,48 @@ Because we expose both React components (for which the markup, including class n
...
+
+   `git checkout -b polaris-release`
+
+3. Migrate all entries from `UNRELEASED.md` to `CHANGELOG.md` ([see example](https://github.com/Shopify/polaris-react/pull/5254/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed))
+
+4. Commit changelog updates
+
+   `git add . && commit -m "Polaris vX.X.X release"`
+
+5. Publish the `polaris-release` branch if it isn't already
```
Comment (aaronccasanova) [inline documentation/Versioning and changelog.md]:
```suggestion
5. Push the `polaris-release` branch to the remote repository

   `git push origin polaris-release` 

```

**Pull request merged**